internal/server/config.go

```diff
@@ -2,31 +2,115 @@ package server
 
 import (
 	"errors"
+	"flag"
 	"fmt"
 	"os"
+	"strconv"
+	"strings"
+
+	"github.com/xChygyNx/metrical/internal/server/types"
 )
 
-type config struct {
-	HostAddr HostPort
+type HostPort struct {
+	Host string
+	Port int
+}
+
+type Config struct {
+	FileStoragePath string
+	HostPort        HostPort
+	StoreInterval   int
+	Restore         bool
+}
+
+var StorageFile = "metrics.json"
+
+func (hp *HostPort) String() string {
+	return fmt.Sprintf("%s:%d", hp.Host, hp.Port)
+}
+
+func (conf *Config) String() string {
+	return fmt.Sprintf(
+		"StoreInterval: %d sec\n"+
+			"FileStoragePath: %s\n"+
+			"Restore: %t\n"+
+			"Host: %s:%d",
+		conf.StoreInterval, conf.FileStoragePath, conf.Restore, conf.HostPort.Host, conf.HostPort.Port)
+}
+
+func (hp *HostPort) Set(value string) error {
+	hostPort := strings.Split(value, ":")
+	numHostPortParts := 2
+	if len(hostPort) != numHostPortParts {
+		errorMsg := "must be value like <Host>:<Port>, got " + value
+		return errors.New(errorMsg)
+	}
+	port, err := strconv.Atoi(hostPort[1])
+	if err != nil {
+		return fmt.Errorf("error in Atoi value of port: %w", err)
+	}
+	hp.Host = hostPort[0]
+	hp.Port = port
+	return nil
+}
+
+func parseFlag() *Config {
+	config := new(Config)
```

Comment:
Так более идиоматично:

```suggestion
	config := &Config{}
```